statsmodels/tsa/statespace/sarimax.py

```diff
@@ -0,0 +1,1941 @@
+"""
+SARIMAX Model
+
+Author: Chad Fulton
+License: Simplified-BSD
+"""
+from __future__ import division, absolute_import, print_function
+
+from warnings import warn
+
+import numpy as np
+from .mlemodel import MLEModel, MLEResults
+from .tools import (
+    companion_matrix, diff, is_invertible, constrain_stationary_univariate,
+    unconstrain_stationary_univariate
+)
+from scipy.linalg import solve_discrete_lyapunov
+from statsmodels.tools.data import _is_using_pandas
+from statsmodels.tsa.tsatools import lagmat
+from statsmodels.tools.decorators import cache_readonly
+
+
+class SARIMAX(MLEModel):
+    r"""
+    Seasonal AutoRegressive Integrated Moving Average with eXogenous regressors
+    model
+
+    Parameters
+    ----------
+    endog : array_like
+        The observed time-series process :math:`y`
+    exog : array_like, optional
+        Array of exogenous regressors, shaped nobs x k.
+    order : iterable or iterable of iterables, optional
+        The (p,d,q) order of the model for the number of AR parameters,
+        differences, and MA parameters. `d` must be an integer
+        indicating the integration order of the process, while
+        `p` and `q` may either be an integers indicating the AR and MA
+        orders (so that all lags up to those orders are included) or else
+        iterables giving specific AR and / or MA lags to include. Default is
+        an AR(1) model: (1,0,0).
+    seasonal_order : iterable, optional
```

Comment:
about `D`: are higher order differencing fully supported or should it be D either zero or one ?